evap/results/tests/test_exporters.py

```diff
@@ -202,6 +202,165 @@ def test_course_type_ordering(self):
+
+        # First, make sure, that the one with 0 voters does not appear normally
+        sheet = self.get_export_sheet(semester, degree, course_types, include_not_enough_voters=False)
+        self.assertEqual(len(sheet.row_values(0)), 2)
+        self.assertEqual(
+            sheet.row_values(0)[1][:-1],
+            enough_voters_evaluation.full_name
+        )
+
+        # Now, check with the option enabled
+        sheet = self.get_export_sheet(semester, degree, course_types, include_not_enough_voters=True)
+        self.assertEqual(len(sheet.row_values(0)), 3)
+        self.assertEqual(
+                {enough_voters_evaluation.full_name, not_enough_voters_evaluation.full_name},
+                {sheet.row_values(0)[1][:-1], sheet.row_values(0)[2][:-1]}
+        )
+
+    def test_no_degree_or_course_type(self):
+        # The exporter asserts, that the degrees / course_types list has something in it,
+        # to avoid an IndexError thrown by xlwt.
+        evaluation = baker.make(Evaluation)
+        with self.assertRaises(AssertionError):
+            ExcelExporter().export(BytesIO(), [evaluation.course.semester], [])
+
+    def test_exclude_single_result(self):
+        degree = baker.make(Degree)
+        evaluation = baker.make(Evaluation, is_single_result=True, state="published", course__degrees=[degree])
+        sheet = self.get_export_sheet(evaluation.course.semester, degree, [evaluation.course.type.id])
+        # There should be no column for the evaluation, only the row description
+        self.assertEqual(len(sheet.row_values(0)), 1)
+
+    def test_exclude_used_but_unanswered(self):
+        degree = baker.make(Degree)
+        evaluation = baker.make(Evaluation, _voter_count=10, _participant_count=10, state="published", course__degrees=[degree])
+        used_questionnaire = baker.make(Questionnaire)
+        used_question = baker.make(Question, type=Question.LIKERT, questionnaire=used_questionnaire)
+        unused_questionnaire = baker.make(Questionnaire)
+        unused_question = baker.make(Question, type=Question.LIKERT, questionnaire=unused_questionnaire)
+        baker.make(RatingAnswerCounter, question=used_question, contribution=evaluation.general_contribution, answer=3, count=100)
+        evaluation.general_contribution.questionnaires.set([used_questionnaire, unused_questionnaire])
+
+        sheet = self.get_export_sheet(evaluation.course.semester, degree, [evaluation.course.type.id])
+        self.assertEqual(sheet.row_values(4)[0], used_questionnaire.name)
+        self.assertEqual(sheet.row_values(5)[0], used_question.text)
+        self.assertNotIn(unused_questionnaire.name, sheet.col_values(0))
+        self.assertNotIn(unused_question.text, sheet.col_values(0))
+
+    def test_degree_course_type_name(self):
+        degree = baker.make(Degree, name_en="Celsius")
+        course_type = baker.make(CourseType, name_en="LetsPlay")
```

Comment:
I think it's funny, however, I'm not sure whether we want to be more serious. @janno42?

Reply:
I'm fine with that in tests. People writing tests do need to have some fun in between because the tests usually aren't :)
We just shouldn't use it in our test_data.json, as it might be confusing for some people because `LetsPlay` is not something users would usually expect for a `CourseType`.